apps/events/views.py

```diff
@@ -213,7 +215,11 @@ def _search_indexed(request, query, filters):
     display_events = set()
 
     for event in events:
-        if event.can_display(request.user):
+        if (event.can_display(request.user) and
+            not(event.is_attendance_event() and
+                filters['attendable'] and
+                event.attendance_event.is_eligible_for_signup(request.user))):
+
```

Comment:
I think you should make a function that returns true or false to make this more readable.